core/domain/subtopic_page_services_test.py

```diff
@@ -62,9 +63,26 @@ def test_get_subtopic_page_from_model(self):
         self.assertEqual(subtopic_page.to_dict(), self.subtopic_page.to_dict())
 
     def test_get_subtopic_page_by_id(self):
-        subtopic_page = subtopic_page_services.get_subtopic_page_by_id(
+        subtopic_page_1 = subtopic_page_services.get_subtopic_page_by_id(
             self.TOPIC_ID, self.subtopic_id)
-        self.assertEqual(subtopic_page.to_dict(), self.subtopic_page.to_dict())
+        self.assertEqual(
+            subtopic_page_1.to_dict(), self.subtopic_page.to_dict())
+        # When the subtopic page with the given subtopic id and topic id
+        # doesn't exist.
+        subtopic_page_2 = subtopic_page_services.get_subtopic_page_by_id(
+            'topic_id', 1, strict=False)
+        self.assertEqual(subtopic_page_2, None)
+
+    def test_get_subtopic_pages_with_ids(self):
```

Comment:
The function being tested would also work for a combination of valid and invalid IDs, right? Add a test case for that. This is an example of good coverage, and also good testing. The test you have currently written achieves coverage, but try to write meaningful tests that cover important corner cases, so that the tests are solid.

Reply:
Also, as you are querying for lists, why do all the tests try single element lists? Try other edge cases like empty lists, multiple elements in lists, same element repeated in lists, etc.

Reply:
Added test for combinations of valid and invalid ids, empty list & list with multi invalid ids.

Reply:
Looks much better now! thanks :)